plugins/out_forward/forward.c

```diff
@@ -1131,6 +1114,56 @@ static void cb_forward_flush(const void *data, size_t bytes,
     FLB_OUTPUT_RETURN(FLB_OK);
 }
 
+static struct flb_config_map config_map[] = {
+    {
+     FLB_CONFIG_MAP_INT, "time_as_integer", NULL,
+     0, FLB_TRUE, offsetof(struct flb_forward_config, time_as_integer),
+     NULL
+    },
+    {
+     FLB_CONFIG_MAP_STR, "shared_key", NULL,
+     0, FLB_TRUE, offsetof(struct flb_forward_config, shared_key),
+     NULL
+    },
+    {
+     FLB_CONFIG_MAP_STR, "self_hostname", "localhost",
+     0, FLB_TRUE, offsetof(struct flb_forward_config, self_hostname),
+     NULL
+    },
+    {
+     FLB_CONFIG_MAP_INT, "empty_shared_key", NULL,
+     0, FLB_TRUE, offsetof(struct flb_forward_config, empty_shared_key),
+     NULL
+    },
+    {
+     FLB_CONFIG_MAP_INT, "send_options", NULL,
+     0, FLB_TRUE, offsetof(struct flb_forward_config,send_options),
+     NULL
+    },
+    {
+     FLB_CONFIG_MAP_INT, "require_ack_response", NULL,
+     0, FLB_TRUE, offsetof(struct flb_forward_config,require_ack_response),
+     NULL
+    },
+    {
+     FLB_CONFIG_MAP_STR, "username", "",
+     0, FLB_TRUE, offsetof(struct flb_forward_config,username),
+     NULL
+    },
+    {
+     FLB_CONFIG_MAP_STR, "password", "",
+     0, FLB_TRUE, 0,
```

Comment:
if we set a default empty string in password "" as the original code states, here we are not processing that value, where is the offsetof ?

Reply:
... and the `upstream` below too. Sorry, I overlooked it.